### PR TITLE
Optimize TransportCongestionControl handling process

### PR DESCRIPTION
**Pull request by johzzy**

### CODE REVIEW

File: worker/include/RTC/TransportCongestionControlServer.hpp [hunk 1]
```diff
@@ -22,65 +21,53 @@ namespace RTC
...
+        return 0u;
+      }
+      virtual void IncomingPacket(uint64_t nowMs, const RTC::RtpPacket* packet) = 0;
+      virtual void TransportConnected() {}
+      virtual void TransportDisconnected() {}
+
+      void SetMaxIncomingBitrate(uint32_t bitrate);
+      void MaySendLimitationRembFeedback();
+
+      static std::unique_ptr<TransportCongestionControlServer::Strategy> Create(
```
Comment (nazar-pc) [inline worker/include/RTC/TransportCongestionControlServer.hpp:36]: I suspect this needs to be `thread_local`

**Comment by nazar-pc** [comment]
Can you write down a few paragraphs about what is done, why and how this optimization positively affects something?

**Comment by johzzy** [comment]
> Can you write down a few paragraphs about what is done, why and how this optimization positively affects something?

This PR splits TWCC and REMB so that they are more independent, also make the execution process more concise and efficient.

Such as, `TransportCongestionControlStrategyREMB` is a`webrtc::RemoteBitrateEstimator::Listener`, And `TransportCongestionControlStrategyTWCC` is a` Timer::Listener`

**Comment by ibc** [comment]
BTW we are writing our own BWE so will remove libwebrtc dep.

**Pull request closed without merging**